GoogleForCreators/web-stories-wp
# Use Interactivity API for frontend functionality

## Summary

This PR makes use of the new Interactivity API to implement the lightbox functionality in the Web Stories block.

On WP versions < 6.5
- There would be no change in the markup of the stories while viewing the block. 
- `packages/stories-lightbox/src/web-stories-lightbox.js` would continue being used to handle the functionality of opening and closing stories.

On WP versions > 6.5
- Interactivity API would be used to open and close stories
- The markup of stories would include Interactivity API directives
- The newly added package `packages/stories-block-view` contains the necessary `state` and `actions`

To learn more about these, please visit [General overview of the Interactivity API](https://developer.wordpress.org/news/2024/04/11/a-first-look-at-the-interactivity-api/#general-overview-of-the-interactivity-api)

## User-facing changes

None. The user won't notice a difference in how the Stories are rendered and opened.

## Testing Instructions

This PR can be tested by following these steps:

1. Ensure that you have WordPress version >= 6.5
2. Add the Web Stories block to a page.
3. View the page and examine the markup of the block.
4. You should see directives (attributes) like `data-wp-interactive`


## Reviews

### Does this PR have a security-related impact?
No.

### Does this PR change what data or activity we track or use?
No.

### Does this PR have a legal-related impact?
No.

## Checklist




---



Fixes #13502
